src/OpenColorIO/ops/lut3d/Lut3DOpCPU_AVX.cpp

```diff
@@ -0,0 +1,328 @@
+{
+    // NOTE: this is a and b are reversed to match sse2 movhlps which is different than unpckhpd
+    return _mm256_castpd_ps(_mm256_unpackhi_pd(_mm256_castps_pd(b), _mm256_castps_pd(a)));
+}
+
+static inline __m256 load2_m128_avx(const float *hi, const float *low)
+{
+    return _mm256_insertf128_ps(_mm256_castps128_ps256(_mm_loadu_ps(low)), _mm_loadu_ps(hi), 1);
+}
+
+#define gather_rgb_avx(src, idx)                                \
+    _mm256_store_si256((__m256i *)indices, idx);                \
+    row0 = load2_m128_avx(src + indices[4], src + indices[0]);  \
+    row1 = load2_m128_avx(src + indices[5], src + indices[1]);  \
+    row2 = load2_m128_avx(src + indices[6], src + indices[2]);  \
+    row3 = load2_m128_avx(src + indices[7], src + indices[3]);  \
+    tmp0 = _mm256_unpacklo_ps(row0, row1);                      \
+    tmp2 = _mm256_unpacklo_ps(row2, row3);                      \
+    tmp1 = _mm256_unpackhi_ps(row0, row1);                      \
+    tmp3 = _mm256_unpackhi_ps(row2, row3);                      \
+    sample_r = movelh_ps_avx(tmp0, tmp2);                       \
+    sample_g = movehl_ps_avx(tmp2, tmp0);                       \
+    sample_b = movelh_ps_avx(tmp1, tmp3)
+
+static inline __m256 fmadd_ps_avx(__m256 a, __m256 b, __m256 c)
+{
+    return  _mm256_add_ps(_mm256_mul_ps(a, b), c);
+}
+
+static inline __m256 blendv_avx(__m256 a, __m256 b, __m256 mask)
+{
+    /* gcc 12.0 to 12.2 don't generate the vblendvps instruction with the -mavx flag.
+       Use inline assembly to force it to.
+       https://gcc.gnu.org/bugzilla/show_bug.cgi?id=106704 */
+#if defined __GNUC__ && __GNUC__ >= 12 && __GNUC_MINOR__ < 3
+    __m256 result;
+    __asm__ volatile("vblendvps %3, %2, %1, %0" : "=x" (result) : "x" (a), "x" (b),"x" (mask):);
+    return result;
+#else
+    return _mm256_blendv_ps(a, b, mask);
+#endif
+}
+
+static inline rgbavec_avx interp_tetrahedral_avx(const Lut3DContextAVX &ctx, __m256 r, __m256 g, __m256 b, __m256 a)
+{
+    AVX_ALIGN(uint32_t indices[8]);
+
+    __m256 x0, x1, x2;
+    __m256 cxxxa;
+    __m256 cxxxb;
+    __m256 mask;
+
+    __m256 tmp0, tmp1, tmp2, tmp3;
+    __m256 row0, row1, row2, row3;
+    __m256 sample_r, sample_g, sample_b;
+
+    rgbavec_avx result;
+
+    __m256 lut_max  = ctx.lutmax;
+    __m256 lutsize  = ctx.lutsize;
+    __m256 lutsize2 = ctx.lutsize2;
+
+    __m256 one_f   = _mm256_set1_ps(1.0f);
+    __m256 four_f  = _mm256_set1_ps(4.0f);
+
+    __m256 prev_r = _mm256_floor_ps(r);
+    __m256 prev_g = _mm256_floor_ps(g);
+    __m256 prev_b = _mm256_floor_ps(b);
+
+    // rgb delta values
+    __m256 d_r = _mm256_sub_ps(r, prev_r);
+    __m256 d_g = _mm256_sub_ps(g, prev_g);
+    __m256 d_b = _mm256_sub_ps(b, prev_b);
+
+    __m256 next_r = _mm256_min_ps(lut_max, _mm256_add_ps(prev_r, one_f));
+    __m256 next_g = _mm256_min_ps(lut_max, _mm256_add_ps(prev_g, one_f));
+    __m256 next_b = _mm256_min_ps(lut_max, _mm256_add_ps(prev_b, one_f));
```

Comment:
Values are scaled and clamped before being passed to `interp_tetrahedral`. 

The trick to clamping the NaNs to zero is to use  `max_ps(value, zero)` before `min_ps(value, max_value)`.  It is also important for the second arg of the `max_ps` intrinsic to be the min/zero arg and not the input pixel value.

Here is a small test program showing it working on every possible float value.
https://godbolt.org/z/3439cvPe8

On I side note, I've noticed using this technique can causes issues when using `sse2neon.h` on clang.  I believe it to be a bug in clang, but haven't reported it to them yet.
https://github.com/DLTcollab/sse2neon/issues/606

Reply:
Oh you're right, it happens in the caller now.  

I looked at your DLTcollab link, we are using different instructions for our min/max implementation in Neon.  Please see this [PR](https://github.com/AcademySoftwareFoundation/OpenColorIO/pull/1775/files) in SSE.h on lines 35-54.

Reply:
I took a quick look at the PR see its using `vmaxnmq_f32` which is the problem. The `fmaxnm` instruction only handles quiet NaNs and not the so call Signalling NaNs. I'll continue this discussion on that PR.

Reply:
I don't think we need/want to suppress Signalling NaNs, do we?  My understanding is that arithmetic operations only generate Quiet NaNs and Signaling NaNs are only set programmatically (e.g. for debugging).

Reply:
I personally think its a good idea to clamp them all to zero regardless in the LUT case. Especially since pixel values are user supplied and being used to calculate memory offsets.